src/sources/util/http_scrape.rs

```diff
@@ -105,6 +106,7 @@ pub(crate) async fn http_scrape<
     inputs: GenericHttpScrapeInputs,
     context_builder: B,
     mut out: SourceSender,
+    http_method: Option<HttpMethod>,
```

Comment:
I think it is a little cleaner to have this not be an `Option`.
The function still resolves to one of the `HttpMethod` enum variants if `None` is passed in (there isn't different behavior taken in the `None` case).

Reply:
So what do you suggest? If I will make just `HttpMethod`, I will need to pass `HttpMethod::Get` in `prometheus` source. Is it ok? If so, I will do it.

Reply:
> I will need to pass HttpMethod::Get in prometheus source

:+1: yes, this is what I'd suggest doing.